### PR TITLE
Update logging docs

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of #7666

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @edward-io @ananthsub

### CODE REVIEW

File: docs/source/extensions/logging.rst
Comment (rohitgr7) [inline docs/source/extensions/logging.rst:153]: don't resolve this one.. need to update once https://github.com/PyTorchLightning/pytorch-lightning/pull/10756 gets merged.

**Comment by rohitgr7** [comment]
here are the docs: https://140072-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html

File: docs/source/extensions/logging.rst
Comment (awaelchli) [inline docs/source/extensions/logging.rst:26]: Can we
1) move the notes to the bottom (seems distracting especially directly in the introduction of the main section here
2) combine the two into one note (could be bullet points)

Alternatively just make it regular text (which I would prefer), because these are items that are anyway immediately visible once they open the logger.

File: docs/source/extensions/logging.rst
Comment (awaelchli) [inline docs/source/extensions/logging.rst:181]: I strongly suggest to either make all titles properly capitalized or not at all. For consistency. For the former, when in doubt, I recommend https://capitalizemytitle.com/

**Review by awaelchli — APPROVED** [review]

File: docs/source/extensions/logging.rst
Comment (tchaton) [inline docs/source/extensions/logging.rst:280]:
```suggestion
If you wish to flush logs to the filesystem at a different steps rate, use the ``flush_logs_every_n_steps`` :class:`~pytorch_lightning.trainer.trainer.Trainer` flag.
```

File: pytorch_lightning/core/lightning.py
Comment (tchaton) [inline pytorch_lightning/core/lightning.py:322]: I believe this one was quite useful. Helped me to understand the default quite fast.

Comment (rohitgr7) [reply pytorch_lightning/core/lightning.py:322]: This is added to the docs now. Because we have more hooks that are not mentioned here, so I moved them inside docs and linked them to the API.

File: docs/source/extensions/logging.rst
Comment (rohitgr7) [reply docs/source/extensions/logging.rst]: thanks! updated!!

File: docs/source/extensions/logging.rst
Comment (rohitgr7) [reply docs/source/extensions/logging.rst]: done!

File: pytorch_lightning/core/lightning.py
Comment (rohitgr7) [reply pytorch_lightning/core/lightning.py:322]: here: https://140072-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html#id4

File: docs/source/extensions/logging.rst
Comment (tchaton) [inline docs/source/extensions/logging.rst:62]: Neat !

File: docs/source/extensions/logging.rst
Comment (tchaton) [inline docs/source/extensions/logging.rst:94]:
```suggestion
    By default, all loggers log to ``os.getcwd()``. You can change the log path by doing ``Trainer(default_root_dir='/your/path/to/save/checkpoints')`` without instantiating a logger.
```

File: docs/source/extensions/logging.rst
Comment (tchaton) [inline docs/source/extensions/logging.rst:118]: Or use:

```py
self.log("performance/acc", acc)
self.log("performance/recall", recall)
```

File: docs/source/extensions/logging.rst
Comment (tchaton) [inline docs/source/extensions/logging.rst:120]:
```suggestion
Depending on where :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method is called, Lightning auto-determines the correct logging mode for you. Of course you can
```

File: docs/source/extensions/logging.rst
Comment (tchaton) [inline docs/source/extensions/logging.rst:133]: Let's list them all. `sync_dist`, etc..

**Review by tchaton — APPROVED** [review]
LGTM ! After comments are resolved.

File: docs/source/extensions/logging.rst
Comment (rohitgr7) [inline docs/source/extensions/logging.rst:116]:
```suggestion
    # or a dict to get multiple metrics on the same plot if the logger supports it
```

File: docs/source/extensions/logging.rst
Comment (rohitgr7) [reply docs/source/extensions/logging.rst:118]: no.. this is different. Your example will create `performance` and will create separate plots for each for acc and recall. The existing example will plot recall and acc on the same plot titled `performance`.

File: docs/source/extensions/logging.rst
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:17]: I liked the statement as it emphasis on our good support for loggers

File: docs/source/extensions/logging.rst
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:40]: Could you hyperlink PyTorch Tensorboard?

File: docs/source/extensions/logging.rst
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:46]:
```suggestion
    # Automatically logs to a directory (by default ``lightning_logs/``)
```

File: docs/source/extensions/logging.rst
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:54]: Isn't it installed by default for Lightning? I don't think the installment command is necessary

File: docs/source/extensions/logging.rst
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:57]:
```suggestion
To visualize tensorboard in a jupyter notebook environment, run the following command in a jupyter cell:
```

File: docs/source/extensions/logging.rst
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:94]:
```suggestion
    By default, all loggers log to ``os.getcwd()``. You can change the logging path using
```

File: docs/source/extensions/logging.rst
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:120]:
```suggestion
Depending on where the :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method is called, Lightning auto-determines
```

File: docs/source/extensions/logging.rst
Comment (kaushikb11) [inline docs/source/extensions/logging.rst:188]:
```suggestion
If you want to log anything that is not a scalar, like histograms, text, images, etc., you may need to use the logger object directly.
```

File: docs/source/extensions/logging.rst
Comment (rohitgr7) [reply docs/source/extensions/logging.rst:17]: we have them right below: https://140783-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html

File: docs/source/extensions/logging.rst
Comment (rohitgr7) [reply docs/source/extensions/logging.rst:40]: we have already listed with links 2 lines above: https://140783-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html

File: docs/source/extensions/logging.rst
Comment (rohitgr7) [inline docs/source/extensions/logging.rst:54]:
```suggestion
```

**Review by kaushikb11 — APPROVED** [review]

File: docs/source/extensions/logging.rst
Comment (carmocca) [inline docs/source/extensions/logging.rst:34]: Should we avoid mentioning `TestTubeLogger`? It's deprecated and unmaintained.

File: docs/source/extensions/logging.rst
Comment (carmocca) [inline docs/source/extensions/logging.rst:369]: Same comment about `TestTubeLogger`

**Review by carmocca — APPROVED** [review]

File: docs/source/extensions/logging.rst
Comment (rohitgr7) [reply docs/source/extensions/logging.rst:34]: yeah, and it will be removed within v1.7? docs will be updated too in the removal PR.

Comment (carmocca) [reply docs/source/extensions/logging.rst:34]: I am personally not a fan of having deprecated features in our docs at all. As an user, I'd rather not learn about this only to later notice it will be removed.

But if you prefer leaving this here until it's finally removed, that's okay. you've got my approval :P

Comment (kaushikb11) [reply docs/source/extensions/logging.rst:34]: I agree with @carmocca, we should remove it.

Comment (rohitgr7) [reply docs/source/extensions/logging.rst:34]: yeah makes sense.. will update :)

**Review by Borda — APPROVED** [review]

**Pull request merged**
